File: LoginRadiusSDK/LRString.m
//
//  NSString+LRString.m
//
//  Copyright © 2016 LoginRadius Inc. All rights reserved.
//

#import "LRString.h"

@implementation NSString (LRString)

- (NSString *) URLDecodedString {
	return [self stringByRemovingPercentEncoding];
}

- (NSString *) URLEncodedString {
    NSMutableCharacterSet *allowedSet = [NSMutableCharacterSet characterSetWithCharactersInString:@":!*();@/&?#[]+$,='%’\""];
    [allowedSet invert];
	return [self stringByAddingPercentEncodingWithAllowedCharacters:[allowedSet copy]];;
}

- (NSString *) capitalizedFirst {
	return [self stringByReplacingCharactersInRange:NSMakeRange(0,1) withString:[[self substringToIndex:1] uppercaseString]];
}

@end

File: Sources/LRString.swift
//
//  LRString.swift
//  LoginRadiusSwift SDK
//
//  Created by Megha Agarwal.
//  Copyright © 2023 LoginRadius Inc. All rights reserved.
//


import Foundation

public extension String {
    
    func URLDecodedString() -> String? {
        return self.removingPercentEncoding
    }
    
    func URLEncodedString() -> String? {
        let allowedSet = CharacterSet(charactersIn: ":!*();@/&?#[]+$,='%’\"").inverted
        return self.addingPercentEncoding(withAllowedCharacters: allowedSet)
    }
    
    func capitalizedFirst() -> String {
        return self.replacingCharacters(in: self.startIndex..<self.index(after: self.startIndex), with: String(self[self.startIndex]).uppercased())
    }
    
}
